### ISSUE TITLE
Compatibility issue with logistics pipes.

### ISSUE THREAD

**Issue by algul** [jaquadro/StorageDrawers#126]
When using polymorphic item sink on drawer/controller items wont be accepted into the drawer if the slot is empty, even when the drawer is locked and labeled to accept it, Technically it works as intended, polymorphic item sink uses items that already exist in the inventory for routing and since the drawer does not have that item, it won't route it there. But I would prefer it to work like ae2 storage bus, where items will get stored there properly.

**Comment by xannor** [comment]
No ones commented, but I think that is by design of the poly sink. It only places items in an inventory that contains similar items, this is an LP thing. I think, as long as you do not have empty spaces, you can just use a default route item sink because that will only route items it can place in an inventory, this is similar to how the ae2 storage bus works.

**Comment by Nagapito** [comment]
Bump on this issue. It might be an LP implementation or it could be that the API is not telling LP the valid items in the drawer when quantity is zero...

**Comment by 1n5aN1aC** [comment]
I've been looking into this issue, And I think it's possible to implement this by a few simple changes in LogisticPipes.

I've actually made those changes locally, and compiled it, and it SEEMS to work, but the changes I've made could possibly cause other issues to surface, so I'm doing some extensive local testing, and on a small-ish server that has several extensive LP networks, connected to StorageDrawers.  So we'll see if any issues arise.

If you'd like a build to test this yourself, just ask, and I can provide a link.  :smiley:
